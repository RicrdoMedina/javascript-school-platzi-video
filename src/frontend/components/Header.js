import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { logoutRequest } from '../actions';
import gravatar from '../utils/gravatar';
import '../assets/styles/components/header.scss';
import logo from '../assets/static/logo-platzi-video-BW2.png';
import iconProfile from '../assets/static/user-icon.svg';

const Header = (props) => {
  const { user } = props;

  const hasUser = Object.keys(user).length > 0 && user.id;

  const handleLogout = () => {
    document.cookie = 'email=';
    document.cookie = 'name=';
    document.cookie = 'id=';
    props.logoutRequest({});
    window.location.href = '/login';
  };

  return (
    <header className="header header--purple">
      <Link to="/">
        <img className="header__img" src={logo} alt="Logo Platzi video" />
      </Link>
      <div className="header__menu">
        <div className="header__menu--profile">
          {hasUser ? (
            <img src={gravatar(user.email)} alt={user.email} />
          ) : (
            <img src={iconProfile} alt="Avatar" />
          )}
          <p>Perfil</p>
        </div>

        {hasUser ? (
          <ul className="header__menu--list">
            <li>
              {/* <Link to="/profile">{user.name}</Link> */}
              <a href="">{user.name}</a>
            </li>
            <li>
              <a href="" onClick={handleLogout}>
                Salir
              </a>
            </li>
          </ul>
        ) : (
          <ul className="header__menu--list">
            <li>
              <Link to="/login">Iniciar Sesion</Link>
            </li>
          </ul>
        )}
      </div>
    </header>
  );
};

Header.propTypes = {
  user: PropTypes.object,
  logoutRequest: PropTypes.func,
};

const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};

const mapDispatchToProps = {
  logoutRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
