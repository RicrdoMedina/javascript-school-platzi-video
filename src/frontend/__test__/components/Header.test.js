import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Header from '../../components/Header';
import ProviderMock from '../../__mocks__/ProviderMock';

describe('<Header/>', () => {
  //Crear Suit colocando el nombre del componente

  test('Header logo image', () => {
    //Probamos que el componente se rendere y que exita un elemento con la clase header__img
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );

    expect(header.find('.header__img')).toHaveLength(1);
  });

  test('Header Snapshot', () => {
    //Crear snapshot del header
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );

    expect(header.toJSON()).toMatchSnapshot();
  });
});
