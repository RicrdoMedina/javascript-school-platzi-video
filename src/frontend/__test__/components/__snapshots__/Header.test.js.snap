// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Header Snapshot 1`] = `
<header
  className="header header--purple"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Logo Platzi video"
      className="header__img"
      src="test-file-stub"
    />
  </a>
  <div
    className="header__menu"
  >
    <div
      className="header__menu--profile"
    >
      <img
        alt="Avatar"
        src="test-file-stub"
      />
      <p>
        Perfil
      </p>
    </div>
    <ul
      className="header__menu--list"
    >
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Iniciar Sesion
        </a>
      </li>
    </ul>
  </div>
</header>
`;
